doc/api/embedding.md

```diff
@@ -160,8 +160,46 @@ int RunNodeInstance(MultiIsolatePlatform* platform,
 }
 ```
 
+## Node-API Embedding
+
+<!--introduced_in=REPLACEME-->
+
+As an alternative, an embedded Node.js can also be fully controlled through
+Node-API. This API supports both C and C++ through [node-addon-api][].
```

Comment:
If we do this, we should probably start adding converters between some Node-API values and V8/Node.js C++ API values (`napi_value` ↔ `Local<Value>`, `napi_env` ↔ `Environment*`/`Isolate*`).

The reason nobody has implemented a Node-API embedder API yet is that as an embedder, you often need fine-grained control over a lot of V8 options, and it’s not very practical to mirror the entire V8/Node.js embedder API to C (and that the Node.js C++ embedder API itself still has some rough edges that should probably be ironed out first).

Reply:
If you need to use raw V8 primitives then you can't use Node-API - this applies to both Node.js -> C++ (binary node addons) and to C++ -> Node.js (libnode). Still this API covers lots of use cases - mostly C++ software that needs to accept JS plugins. It does not cover Electron which has very particular needs.
Allowing for more fine-grained control over the initialisation is of course something that should be considered.

Reply:
> If you need to use raw V8 primitives then you can't use Node-API

Right, that’s why I’m bringing this up.

Reply:
Should you be able to mix V8 primitives and Node-API is another topic. If you take the ABI compatibility out of Node-API, there remains little use for it. However `node-addon-api` is a truly higher level API and it is much more practical to use from C++ then raw V8. Maybe there is a use case for software which has 95% `node-addon-api` and 5% custom V8 code. But this is something that goes beyond this PR - and it probably can be used both by embedders and addon authors.

Reply:
I believe Cloudflare ([service](https://github.com/cloudflare/worker-emscripten-template)) Workers will need [converters](https://markethistory.quora.com/What-is-the-relationship-between-node-js-and-C-1). Is this a Private API thing?